### PR TITLE
Add MviViewModel with StateFlow

### PR DESCRIPTION
**Pull request by gorrotowi**
Include ViewModel with Intents and TodoState for the View layer also was added an example in mvi_example of how to use it

[Implement View layer](https://distillery.atlassian.net/browse/AAG-5?atlOrigin=eyJpIjoiODgwYmE1MDQ3M2IwNGQ0YjljMGE0MWE1ZjBjN2M4YzIiLCJwIjoiaiJ9)

**Type of change:**
- New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**
- Pending Unit Testing

**Checklist:**
- [x] This PR follows [guidelines](https://github.com/DistilleryTech/android-arch-blueprints/wiki/Pull-Request-Process)
- [x] I have performed a self-review of my own code
- [ ] I documented all new code and updated the external docs if necessary
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: mvi_example/src/main/java/com/distillery/android/blueprints/mvi/MviActivity.kt [hunk 1]
```diff
@@ -1,11 +1,58 @@
 package com.distillery.android.blueprints.mvi
 
 import android.os.Bundle
+import android.widget.Button
 import androidx.appcompat.app.AppCompatActivity
+import com.distillery.android.blueprints.mvi.todo.repo.model.TodoModel
+import com.distillery.android.blueprints.mvi.todo.state.TodoState
+import com.distillery.android.blueprints.mvi.todo.viewmodel.TodoIntent
```
Comment (distillery-alexander-ageychenko) [inline mvi_example/src/main/java/com/distillery/android/blueprints/mvi/MviActivity.kt:8]: Note: `Unresolved reference: TodoIntent`

File: mvi/src/main/java/MviIntent.kt [hunk 2]
```diff
@@ -0,0 +1 @@
+interface MviIntent
```
Comment (distillery-alexander-ageychenko) [inline mvi/src/main/java/MviIntent.kt:1]: I think there should be some package mentioned, could you check?

File: mvi_example/src/main/java/com/distillery/android/blueprints/mvi/MviActivity.kt (hunk 1 above)
Comment (gorrotowi) [reply mvi_example/src/main/java/com/distillery/android/blueprints/mvi/MviActivity.kt]: Fixed :)

File: mvi/src/main/java/MviIntent.kt (hunk 2 above)
Comment (gorrotowi) [reply mvi/src/main/java/MviIntent.kt]: I was working in the root directory from the module, I already add the correct package

**Pull request merged**